src/libraries/System.ComponentModel.TypeConverter/ref/System.ComponentModel.TypeConverter.cs

```diff
@@ -1055,7 +1057,8 @@ public abstract partial class PropertyDescriptor : System.ComponentModel.MemberD
         public abstract bool IsReadOnly { get; }
         public abstract System.Type PropertyType { get; }
         public System.ComponentModel.DesignerSerializationVisibility SerializationVisibility { get { throw null; } }
-        public virtual bool SupportsChangeEvents { get { throw null; } }
+        public virtual bool SupportsChangeEvents { [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("The built-in EventDescriptor implementation uses Reflection which requires unreferenced code.")] get { throw null; } }
+        [System.Diagnostics.CodeAnalysis.RequiresUnreferencedCodeAttribute("The built-in EventDescriptor implementation uses Reflection which requires unreferenced code.")]
```

Comment:
Why did we need to make these changes to `SupportsChangeEvents`, `AddValueChanged` and `RemoveValueChanged`?

Reply:
SupportsChangeEvents, AddValueChanged and RemoveValueChanged call IPropChangedEventValue.get which calls: TypeDescriptor.GetEvents(typeof(INotifyPropertyChanged)) and that calls into ICustomTypeDescriptor.GetEvents() which is marked as RUC. I'm not sure about this 100% since the type is known but at the same time even when I called GetEvents() in the tests when I marked the derived test type with DAM(all) it still was trimming the events

Reply:
What do you think about changing `DbConnectionStringBuilder.GetEvents()` method to follow the same pattern as above - call `this.GetType()` (with a comment) and add a suppression? Then we don't need to change the Events stuff at all.

@krwq @vitek-karas - Thoughts?

Reply:
I've just tried GetEvents and it seems it works fine when you add GetType. Will also try GetProperties

Reply:
properties also working, PR updated, PTAL

Reply:
Given the current limitations this looks good. Ideally there would be an overload on `TypeDescriptor` which can take the type as well as an instance, but that's new public API which is tricky...